ReplayKitExample/ReplayKitExample/ViewController.swift

```diff
@@ -0,0 +1,401 @@
+
+    static let kStartBroadcastButtonTitle = "Start Broadcast"
+    static let kInProgressBroadcastButtonTitle = "Broadcasting"
+    static let kStopBroadcastButtonTitle = "Stop Broadcast"
+    static let kStartConferenceButtonTitle = "Start Conference"
+    static let kStopConferenceButtonTitle = "Stop Conference"
+    static let kRecordingAvailableInfo = "Ready to share the screen in a Broadcast (extension), or Conference (in-app)."
+    static let kRecordingNotAvailableInfo = "ReplayKit is not available at the moment. Another app might be recording, or AirPlay may be in use."
+
+    // An application has a much higher memory limit than an extension. You may choose to deliver full sized buffers instead.
+    static let kDownscaleBuffers = false
+
+    deinit {
+        NotificationCenter.default.removeObserver(self)
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        broadcastButton.setTitle(ViewController.kStartBroadcastButtonTitle, for: .normal)
+        conferenceButton?.setTitle(ViewController.kStartConferenceButtonTitle, for: .normal)
+        broadcastButton.layer.cornerRadius = 4
+        conferenceButton?.layer.cornerRadius = 4
+
+        // The setter fires an availability changed event, but we check rather than rely on this implementation detail.
+        RPScreenRecorder.shared().delegate = self
+        checkRecordingAvailability()
+
+        NotificationCenter.default.addObserver(forName: NSNotification.Name.UIScreenCapturedDidChange, object: UIScreen.main, queue: OperationQueue.main) { (notification) in
+            if self.broadcastPickerView != nil && self.screenTrack == nil {
+                let isCaptured = UIScreen.main.isCaptured
+                let title = isCaptured ? ViewController.kInProgressBroadcastButtonTitle : ViewController.kStartBroadcastButtonTitle
+                self.broadcastButton.setTitle(title, for: .normal)
+                self.conferenceButton?.isEnabled = !isCaptured
+                isCaptured ? self.spinner.startAnimating() : self.spinner.stopAnimating()
+            }
+        }
+
+        // Use RPSystemBroadcastPickerView when available (iOS 12+ devices).
+        // TODO: Use #if targetEnvironment(simulator) after upgrading the examples to Swift 4.2.
+        #if arch(arm64)
+        if #available(iOS 12.0, *) {
+            // Swap the button for an RPSystemBroadcastPickerView.
+            let pickerView = RPSystemBroadcastPickerView(frame: CGRect(x: 0,
+                                                                       y: 0,
+                                                                   width: view.bounds.width,
+                                                                  height: 80))
+            pickerView.preferredExtension = "com.twilio.ReplayKitExample.BroadcastVideoExtension"
+            view.addSubview(pickerView)
+
+            self.broadcastPickerView = pickerView
+            broadcastButton.isEnabled = false
+        }
```

Comment:
I agree, right now the UI is a bit swiss cheese. It could be made much better with some formal definition of state, or at least some shared methods to do the transitions.